### PR TITLE
fun accessors for initial amounts and concentrations

### PR DESCRIPTION
**Pull request by exaexa**
cc @paulflang

### CODE REVIEW

**Review by paulflang — COMMENTED** [review]
I think this could work with a minor modification to SBMLToolkit and 2 minor modifications here:

- Shouldn't it be `m.species` instead of `m.metabolites`?
- Since the m.metabolites is a `Dict` (i.e. unordered), I think the return vector will be in some weird order. Can we return a vector of `metabolite.id` => initial_amount `Pairs` (that type would be identical to what MTK needs, except that I have to convert the `metabolite.id` from `String` to a time dependent Symbolics FnType).

**Comment by paulflang** [comment]
OK. A second thought on the types I had when thinking about how to do the analogous four kinetic_math. Do you think `initial_amount` should return the same type as `model.metabolites`, i.e. a `Dict(String, Species)`? Or shall we do `Dict(String, Real)` instead of the Vector of Pairs I suggested above?

Similarly do you think the `to_extensive_math()` function (or whatever more appropriate name we come up with) should return the same type as `model.reactions`, i.e. a `Dict(String, Reaction)`? Or shall we do `Dict(String, Math)` or `Vector{Pair{String,Math}}`?

**Comment by anandijain** [comment]
```julia
julia> subtypes(AbstractDict)
9-element Vector{Any}:
 Base.EnvDict
 Base.ImmutableDict
 Base.Iterators.Pairs
 Dict
 IdDict
 OrderedCollections.LittleDict
 OrderedCollections.OrderedDict
 Test.GenericDict
 WeakKeyDict
```
It looks like a Pair iterator subtypes AbstractDict, so you may be able to keep it general

**Comment by exaexa** [comment]
OK I guess I can return a Dict, should be better. I'll look into returning actual generators later, with all other types.

File: test/ecoli_flux.jl [hunk 1]
```diff
@@ -18,7 +18,9 @@ end
 
     @test typeof(mdl) == Model
 
-    @test_logs (:error, "SBML reported error: File unreadable.") @test_throws AssertionError readSBML(sbmlfile * ".does.not.really.exist")
+    @test_logs (:error, "SBML reported error: File unreadable.") @test_throws AssertionError readSBML(
+        sbmlfile * ".does.not.really.exist",
```
Comment (anandijain) [inline test/ecoli_flux.jl:22]: nitpick but why is the error message `.does.not.really.exist` with the periods?

Comment (exaexa) [reply test/ecoli_flux.jl:22]: that's not the mesage.

Comment (anandijain) [reply test/ecoli_flux.jl:22]: ok u can resolve this then

Comment (anandijain) [reply test/ecoli_flux.jl:22]: oh yeah totally misread that

**Comment by paulflang** [comment]
In the interest of time it might make sense to return a Dict rather than a generator?

**Comment by anandijain** [comment]
Yeah can we get this in soon, i really don't mind Dict

**Comment by exaexa** [comment]
@paulflang @anandijain the functions now return generators, you can easily get any type you want. See docs of `initial_amounts` for an example with dictionaries.

**Pull request merged**